github/github.go

```diff
@@ -880,6 +880,32 @@ func cloneRequest(r *http.Request) *http.Request {
 	return r2
 }
 
+// formatRateLimitResetDuration formats d to look like "[rate reset in 2s]" or
+// "[rate reset in 87m02s]" for the positive durations. And like "[rate limit was reset 87m02s ago]"
+// for the negative cases.
+func formatRateLimitResetDuration(d time.Duration) string {
+	isNegative := d < 0
+	var timeString string
```

Comment:
In general, it's a good practice not to do unnecessary things, which includes declaring variables way earlier than necessary.

In this case, `var timeString string` is declared here, but not used/needed until 7 lines below.

Let's move it so it's declared just before it's used, before `if minutes > 0` line.

That should improve readability of this function.